extension/partiql-extension-ion/src/lib.rs

```diff
@@ -0,0 +1,211 @@
+mod common;
+pub mod decode;
+pub mod encode;
+
+pub use common::Encoding;
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::decode::{IonDecodeResult, IonDecoderBuilder, IonDecoderConfig};
+    use crate::encode::{IonEncodeError, IonEncoderBuilder, IonEncoderConfig};
+    use itertools::Itertools;
+
+    use partiql_value::{partiql_bag, partiql_list, partiql_tuple, DateTime, Value};
+    use rust_decimal_macros::dec;
+    use std::num::NonZeroU8;
+
+    pub fn decode_ion(contents: &str, encoding: Encoding) -> IonDecodeResult {
```

Comment:
Doesn't need to be `pub`

```suggestion
   fn decode_ion(contents: &str, encoding: Encoding) -> IonDecodeResult {
```